ddcz/html.py

```diff
@@ -250,3 +251,35 @@ def encode_valid_html(entity_string):
                 continue
 
     return encoded_safe_string
+
+
+class HtmlChecker(HTMLParser):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.opened_tags = []
+
+    def handle_starttag(self, tag, attrs):
+        if tag not in ['br']:
+            self.opened_tags.append(tag)
+
+    def handle_endtag(self, tag):
+        if len(self.opened_tags) == 0:
+            raise HtmlTagMismatchException(
+                f"Attempt to close tag {tag} when no tags are open"
+            )
+
+        if self.opened_tags[-1] != tag:
+            raise HtmlTagMismatchException(
+                f"Unclosed tag {self.opened_tags[-1]} when tag {tag} encountered"
+            )
+        del self.opened_tags[-1]
+
+
+class HtmlTagMismatchException(Exception):
+    """ Pairing of the tags is not done properly """
```

Comment:
This will be my lack of knowledge of Python, but is this really correctly written?

Reply:
Yes, those are [docstrings](https://www.python.org/dev/peps/pep-0257/)